akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExample5b.scala

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.PersistentBehaviors5
+
+/*
+API experiment with factory for command and event handler
+- same as AccountExample5 but with explicit types rather than type alias
+- no enclosing class
+- nothing special at all, using defs for different command handlers
+- using PersistentBehaviors5
+*/
+
+object AccountExample5b {
+
+  object AccountEntity {
+    sealed trait AccountCommand
+    case object CreateAccount extends AccountCommand
+    case class Deposit(amount: Double) extends AccountCommand
+    case class Withdraw(amount: Double) extends AccountCommand
+    case object CloseAccount extends AccountCommand
+
+    sealed trait AccountEvent
+    case object AccountCreated extends AccountEvent
+    case class Deposited(amount: Double) extends AccountEvent
+    case class Withdrawn(amount: Double) extends AccountEvent
+    case object AccountClosed extends AccountEvent
+
+    sealed trait Account {
+      def applyEvent(event: AccountEvent): Account
+    }
+    case object EmptyAccount extends Account {
+      override def applyEvent(event: AccountEvent): Account = event match {
+        case AccountCreated ⇒ OpenedAccount(0.0)
+        case _              ⇒ throw new IllegalStateException(s"unexpected event [$event] in state [EmptyAccount]")
+      }
+    }
+    case class OpenedAccount(balance: Double) extends Account {
+      override def applyEvent(event: AccountEvent): Account = event match {
+        case Deposited(amount) ⇒ copy(balance = balance + amount)
+        case Withdrawn(amount) ⇒ copy(balance = balance - amount)
+        case AccountClosed     ⇒ ClosedAccount
+        case _                 ⇒ throw new IllegalStateException(s"unexpected event [$event] in state [OpenedAccount]")
+      }
+    }
+    case object ClosedAccount extends Account {
+      override def applyEvent(event: AccountEvent): Account =
+        throw new IllegalStateException(s"unexpected event [$event] in state [ClosedAccount]")
+    }
+
+    private def initialHandler(cmd: AccountCommand): Effect[AccountEvent, Account] =
+      cmd match {
+        case CreateAccount ⇒ Effect.persist(AccountCreated)
+        case _             ⇒ Effect.unhandled
+      }
+
+    private def openedAccountHandler(acc: OpenedAccount, cmd: AccountCommand): Effect[AccountEvent, Account] = {
```

Comment:
+1 much prefer these as methods. Writing the types out does not seem verbose and no new concepts, just methods.